dali/base/dameta.cpp

```diff
@@ -0,0 +1,351 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+#include "dameta.hpp"
+#include "dadfs.hpp"
+#include "dautils.hpp"
+
+
+//More to a more central location
+IPropertyTree * queryHostGroup(const char * name)
+{
+    if (isEmptyString(name))
+        return nullptr;
+    VStringBuffer xpath("storage/hostGroups[@name='%s']", name);
+    IPropertyTree & global = queryGlobalConfig();
+    return global.queryPropTree(xpath);
+}
+
+IPropertyTree * queryStoragePlane(const char * name)
+{
+    VStringBuffer xpath("storage/planes[@name='%s']", name);
+    IPropertyTree & global = queryGlobalConfig();
+    return global.queryPropTree(xpath);
+}
+
+//Cloned for now - export and use from elsewhere
+
+static void copyPropIfMissing(IPropertyTree & target, const char * targetName, IPropertyTree & source, const char * sourceName)
+{
+    if (source.hasProp(sourceName) && !target.hasProp(targetName))
+    {
+        if (source.isBinary(sourceName))
+        {
+            MemoryBuffer value;
+            source.getPropBin(sourceName, value);
+            target.setPropBin(targetName, value.length(), value.toByteArray());
+        }
+        else
+            target.setProp(targetName, source.queryProp(sourceName));
+    }
+}
+
+static void copySeparatorPropIfMissing(IPropertyTree & target, const char * targetName, IPropertyTree & source, const char * sourceName)
+{
+    //Legacy - commas are quoted if they occur in a separator list, so need to remove the leading backslashes
+    if (source.hasProp(sourceName) && !target.hasProp(targetName))
+    {
+        StringBuffer unquoted;
+        const char * text = source.queryProp(sourceName);
+        while (*text)
+        {
+            if ((text[0] == '\\') && (text[1] == ','))
+                text++;
+            unquoted.append(*text++);
+        }
+        target.setProp(targetName, unquoted);
+    }
+}
+
+//====================================================================================================================
+
+
+
+class LogicalFileResolver
+{
+public:
+    LogicalFileResolver(IUserDescriptor * _user, ResolveOptions _options)
+    : options(_options)
```

Comment:
_user unused ? or should stored in member user?

Reply:
well spotted.